storage_service/common/middleware.py

```diff
@@ -16,6 +16,36 @@
     EXEMPT_URLS += [compile(expr) for expr in settings.LOGIN_EXEMPT_URLS]
 
 
+class AuditLogMiddleware(object):
+    """Add X-Username header with authenticated user to each response.
+
+    This provides a mechanism for including the username of the
+    authenticated user for each request/response in the nginx access
+    logs for auditing purposes.
+
+    To make use of this header in the nginx conf, add the following to
+    `log_format`:
+
+    user=$upstream_http_x_username
+
+    For more information on this middleware's use as part of an
+    Archivematica auditing solution, see the README for auditmatica:
+    https://github.com/artefactual-labs/auditmatica.
+    """
+
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        response = self.get_response(request)
+
+        username = None
+        if request.user.is_authenticated:
+            username = request.user.get_username()
+        response["X-Username"] = username
```

Comment:
```suggestion
        if request.user.is_authenticated:
            response["X-Username"] = username        
```
I guess this is both a suggestion and a question. If the user isn't authenticated, and the user isn't returned from get_username do we need to modify the header and send anything? You'll see if you do a `HEAD` request you'll get `None` in this field. Depending on your perspective this might be seen as unnecessary, but it might also be confusing. 

What do you think? 

(Although I see in your tests you're testing `None` explicitly on logout.)

Reply:
I want to do a bit of testing to make sure I understand the impacts, but I think you're right